pkg/internal/cmd/options/options.go

```diff
@@ -17,47 +17,123 @@ limitations under the License.
 package options
 
 import (
+	"flag"
+	"fmt"
+
 	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
 	// to ensure that exec-entrypoint and run can make use of them.
-	"flag"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
 
 	"github.com/go-logr/logr"
 	"github.com/spf13/cobra"
-	_ "k8s.io/client-go/plugin/pkg/client/auth"
-	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"github.com/spf13/pflag"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/klog/v2"
+	"k8s.io/klog/v2/klogr"
 )
 
+// Options are the main options for the policy-approver. Populated via
+// processing command line flags.
 type Options struct {
+	// logLevel is the verbosity level the driver will write logs at.
+	logLevel string
+
+	// kubeConfigFlags is used for generating a Kubernetes rest config via CLI
+	// flags.
+	kubeConfigFlags *genericclioptions.ConfigFlags
+
+	// MetricsAddress is the TCP address for exposing HTTP Prometheus metrics
+	// which will be served on the HTTP path '/metrics'. The value "0" will
+	// disable exposing metrics.
 	MetricsAddress string
-	ProbeAddress   string
 
+	// ReadyzAddress is the TCP address for exposing the HTTP readiness probe
+	// which will be served on the HTTP path '/readyz'.
+	ReadyzAddress string
+
+	// ApproveWhenNoPolicies configures policy-approver to approve all
+	// CertificateRequests if no CertificateRequestPolicies resources exist.
+	ApproveWhenNoPolicies bool
+
+	// LeaderElectionNamespace is the namespace in which leader election should
+	// be leased in to form leader election.
 	LeaderElectionNamespace string
 
-	zapOptions *zap.Options
```

Comment:
Curious why you decided to not use zap?

Reply:
Klog is what we use for other cert-manager projects, and also integrates with the kubernetes libraries. Setting verbose log levels will make the Kubernetes libraries and HTTP clients also verbose.